### PR TITLE
fixed link deletion

### LINKED ISSUE #949: Private share not deleted using bulk delete with new template
I installed https://github.com/shaarli/Shaarli/releases/tag/v0.9.1 on a new server today and I noticed a problem with the new template : The selected private share checked are not deleted by bulk delete.
I selected one link and clicked on delete :
image
Then I click sur ok :
image
Nothing happened

It may be connected to these php errors I found in the log
```
[Sun Aug 27 18:29:05.620534 2017] [:error] [pid 12426] [client 127.0.0.1:33272] PHP Notice:  Undefined variable: link in /var/www/shaarli/index.php on line 1340
[Sun Aug 27 18:29:05.621621 2017] [:error] [pid 12426] [client 127.0.0.1:33272] PHP Notice:  Undefined variable: link in /var/www/shaarli/index.php on line 1351
```

### PR DESCRIPTION
**Pull request by thewilli**
fixes bulk deletion of links (see commit message)

fixes #949

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -1320,10 +1320,17 @@ function renderPage($conf, $pluginManager, $LINKSDB, $history)
             die('Wrong token.');
         }
 
-        if (strpos($_GET['lf_linkdate'], ' ') !== false) {
-            $ids = array_values(array_filter(preg_split('/\s+/', escape($_GET['lf_linkdate']))));
+        $ids = trim($_GET['lf_linkdate']);
+        if (strpos($ids, ' ') !== false) {
+            // multiple, save-separated ids provided
```
Comment (nodiscc) [inline index.php]: s/save/space ?

Comment (thewilli) [reply index.php]: @nodiscc yes, sorry, just fixed it

File: index.php [hunk 2]
```diff
@@ -1320,10 +1320,17 @@ function renderPage($conf, $pluginManager, $LINKSDB, $history)
...
+            // multiple, space-separated ids provided
+            $ids = array_values(array_filter(preg_split('/\s+/', escape($ids))));
         } else {
-            $ids = [$_GET['lf_linkdate']];
+            // only a single id provided
+            $ids = [$ids];
+        }
+        // assert at least one id is given
+        if(!count($ids)){
+            die('no id provided');
```
Comment (ArthurHoaro) [inline index.php]: Is the hard die necessary here? Maybe just wrap your `if (count())` around the foreach and saves.

Comment (thewilli) [reply index.php]: > Is the hard die necessary here? Maybe just wrap your if (count()) around the foreach and saves?

imho it is necessary. The user requested a page whose only purpose it is to delete links. This invocation without any id shouldn't happen at all and can only occur if something went wrong. So if this `die()` ever gets called, it should indicate an issue one should look after.

Comment (ArthurHoaro) [reply index.php]: Alright, anyway it *shouldn't* happen.

**Pull request merged**